packages/emitter-framework/test/typescript/components/value-expression.test.tsx

```diff
@@ -0,0 +1,236 @@
+import { Output, render } from "@alloy-js/core";
+import { dedent } from "@alloy-js/core/testing";
+import { SourceFile } from "@alloy-js/typescript";
+import { EnumValue, Model, Namespace, Numeric, NumericValue, Value } from "@typespec/compiler";
+import { $ } from "@typespec/compiler/experimental/typekit";
+import { assert, beforeAll, describe, expect, it } from "vitest";
+import { ValueExpression } from "../../../src/typescript/components/value-expression.js";
+import { getProgram, initEmptyProgram } from "../test-host.js";
+
+beforeAll(async () => {
+  await initEmptyProgram();
+});
+
+it("renders strings", async () => {
+  const value = $.value.createString("test");
+
+  await testValueExpression(value, `"test"`);
+});
+
+describe("numeric values", () => {
+  it("renders integers", async () => {
+    const value = $.value.createNumeric(42);
+
+    await testValueExpression(value, `42`);
+  });
+
+  it("renders decimals", async () => {
+    const value = $.value.createNumeric(42.5);
+
+    await testValueExpression(value, `42.5`);
+  });
+
+  it("renders bigints", async () => {
+    const digits = "1234567890123456789012345678901234567890";
+    const value: NumericValue = {
+      entityKind: "Value",
+      valueKind: "NumericValue",
+      value: Numeric(digits),
+    } as NumericValue;
+
+    await testValueExpression(value, `${digits}n`);
+  });
+
+  it("throws on invalid numbers", async () => {
+    const digits = "123456789123456789.112233445566778899";
+    const value: NumericValue = {
+      entityKind: "Value",
+      valueKind: "NumericValue",
+      value: Numeric(digits),
+    } as NumericValue;
+
+    await expect(testValueExpression(value, ``)).rejects.toThrow("BigInt value must be an integer");
+  });
+});
+
+it("renders booleans", async () => {
+  const value = $.value.createBoolean(true);
+  await testValueExpression(value, `true`);
+});
+
+it("renders nulls", async () => {
+  const value = {
+    entityKind: "Value",
+    valueKind: "NullValue",
+    value: null,
+  } as Value;
+  await testValueExpression(value, `null`);
+});
+
+it("renders empty arrays", async () => {
+  // Can be replaced with with TypeKit once #6976 is implemented
+  const value = {
+    entityKind: "Value",
+    valueKind: "ArrayValue",
+    values: [],
+  } as unknown as Value;
+  await testValueExpression(value, `[]`);
+});
+
+it("renders arrays with mixed values", async () => {
+  // Can be replaced with with TypeKit once #6976 is implemented
+  const value = {
+    entityKind: "Value",
+    valueKind: "ArrayValue",
+    values: [$.value.createString("foo"), $.value.createNumeric(42), $.value.createBoolean(true)],
+  } as Value;
+  await testValueExpression(value, `["foo", 42, true]`);
+});
+
+it("renders scalars", async () => {
+  const program = await getProgram(`
```

Comment:
If you use `createEmitterFrameworkTestRunner()` directly you can put `@test` on things you're interested in and it will return them for you, e.g. you could do here:

```ts
const runner = await createEmitterFrameworkTestRunner();
const { minDate } = runner.compile(...code...)
```